### PR TITLE
Added SideConfiguration and INBTSaveable to the API. Changed some IPipeTile to IInjectable. Tweaked the auto-eject method. Closes #315.

### LINKED ISSUE #315: Connection Bug
Normal BC pipes connect to configurable dusters only when the side configuration is INPUT or BOTH.

Maybe only allow the auto-ejecting only for extraction type pipes?

### PR DESCRIPTION
**Pull request by iTitus**

### CODE REVIEW

File: src/main/java/buildcraftAdditions/api/networking/ISyncObject.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
  */
 public interface ISyncObject {
 
-	public ByteBuf writeToByteBuff(ByteBuf buf);
+	void writeToByteBuff(ByteBuf buf);
 
-	public ByteBuf readFromByteBuff(ByteBuf buf);
+	void readFromByteBuff(ByteBuf buf);
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/api/networking/ISyncObject.java]: it's clear you didn't test this, this won't work at all, if you don't return the bytebuff it will be reading wrong values

Comment (iTitus) [inline src/main/java/buildcraftAdditions/api/networking/ISyncObject.java]: ByteBufs are mutable. The internal counter counts.

Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/api/networking/ISyncObject.java]: it should work that way but when i was testing this earlier it didn't work at all, it was reading incorrect data as if it started reading from the beginning of the bytebuf each time

Comment (iTitus) [inline src/main/java/buildcraftAdditions/api/networking/ISyncObject.java]: Strange, works for me

File: src/main/java/buildcraftAdditions/api/configurableOutput/EnumPriority.java [hunk 2]
```diff
@@ -26,7 +26,7 @@ private EnumPriority(String name, int color) {
 	}
 
 	public String getName() {
-		return Utils.localize("priority." + name);
+		return ("" + StatCollector.translateToLocal("priority." + name)).trim();
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/api/configurableOutput/EnumPriority.java]: why this change?

Comment (iTitus) [inline src/main/java/buildcraftAdditions/api/configurableOutput/EnumPriority.java]: The API should NEVER access classes in the main mod.

Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/api/configurableOutput/EnumPriority.java]: oh right, forgot that this was in the api for a sec

**Comment by iTitus** [comment]
Finished,
